src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -75,6 +79,103 @@ public NetworkModificationService(NetworkStoreService networkStoreService, Netwo
+                modifications = energiseLineEnd(networkUuid, lineId, Branch.Side.ONE);
+                break;
+            case "energiseEndTwo":
+                modifications = energiseLineEnd(networkUuid, lineId, Branch.Side.TWO);
+                break;
+            default:
+                throw new IllegalStateException("Unexpected value: " + lineStatus);
+        }
+        return modifications;
+    }
+
+    public Flux<ElementaryModificationInfos> lockoutLine(UUID networkUuid, String lineId) {
+        return getNetwork(networkUuid)
+                .filter(network -> network.getLine(lineId) != null)
+                .switchIfEmpty(Mono.error(new NetworkModificationException(LINE_NOT_FOUND, lineId)))
+                .flatMapIterable(network -> doAction(network, networkUuid, () -> {
+                    if (disconnectLineBothSides(network, lineId)) {
+                        network.getLine(lineId).newExtension(BranchStatusAdder.class).withStatus(BranchStatus.Status.PLANNED_OUTAGE).add();
+                    } else {
+                        throw new NetworkModificationException(MODIFICATION_ERROR, "Unable to disconnect both line ends");
+                    }
+                }
+                ));
+    }
+
+    public Flux<ElementaryModificationInfos> tripLine(UUID networkUuid, String lineId) {
+        return getNetwork(networkUuid)
+                .filter(network -> network.getLine(lineId) != null)
+                .switchIfEmpty(Mono.error(new NetworkModificationException(LINE_NOT_FOUND, lineId)))
+                .flatMapIterable(network -> doAction(network, networkUuid, () -> {
+                    if (disconnectLineBothSides(network, lineId)) {
+                        network.getLine(lineId).newExtension(BranchStatusAdder.class).withStatus(BranchStatus.Status.FORCED_OUTAGE).add();
+                    } else {
+                        throw new NetworkModificationException(MODIFICATION_ERROR, "Unable to disconnect both line ends");
+                    }
+                }
+                ));
+    }
+
+    public Flux<ElementaryModificationInfos> energiseLineEnd(UUID networkUuid, String lineId, Branch.Side side) {
+        return getNetwork(networkUuid)
+                .filter(network -> network.getLine(lineId) != null)
+                .switchIfEmpty(Mono.error(new NetworkModificationException(LINE_NOT_FOUND, lineId)))
+                .flatMapIterable(network -> doAction(network, networkUuid, () -> {
+                    Terminal terminalToConnect = network.getLine(lineId).getTerminal(side);
+                    boolean isTerminalToConnectConnected = terminalToConnect.isConnected() || terminalToConnect.connect();
+                    Terminal terminalToDisconnect = network.getLine(lineId).getTerminal(side == Branch.Side.ONE ? Branch.Side.TWO : Branch.Side.ONE);
+                    boolean isTerminalToDisconnectDisconnected = !terminalToDisconnect.isConnected() || terminalToDisconnect.disconnect();
```

Comment:
I don't understand this piece of code, why to you try to disconnect the other side?

Reply:
ok ok ok